### PR TITLE
Add `CX`, `CY` and `CZ` to Type1, Type3 target packages.

### PR DESCRIPTION
**Pull request by swernli**
This simplifies the use of controlled Pauli gates to allow for programs that make use of singly controlled gates (or CCNOT) to have reasonable decompositoins without incurring array creation or using the controlled specialization. This leads to better compatibility with the draft spec for the QIR base profile.

### CODE REVIEW

**Comment by swernli** [comment]
Note that this is going into a feature branch for now as part of a larger effort to apply simplifications to the target packages and ease their use across different QIR profiles.

This is related to https://github.com/microsoft/QuantumLibraries/issues/477, but does not necessarily remove the need for considering that proposal.

File: src/Simulation/TargetDefinitions/Decompositions/CX.qs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+namespace Microsoft.Quantum.Canon {
```
Comment (cgranade) [inline src/Simulation/TargetDefinitions/Decompositions/CX.qs:1]: Do we need a copyright header here?

File: src/Simulation/TargetDefinitions/Decompositions/CYFromCNOT.qs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+namespace Microsoft.Quantum.Canon {
```
Comment (cgranade) [inline src/Simulation/TargetDefinitions/Decompositions/CYFromCNOT.qs:1]: Similarly here.

File: src/Simulation/TargetDefinitions/Decompositions/CZFromSinglyControlled.qs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    /// Equivalent to:
+    /// ```qsharp
+    /// Controlled Z([control], target);
+    /// ```
+    operation CZ(control : Qubit, target : Qubit) : Unit {
+        body (...) {
+            ApplyControlledZ(control, target);
+        }
+        controlled (ctls, ...) {
+            if (Length(ctls) == 0) {
```
Comment (cgranade) [inline src/Simulation/TargetDefinitions/Decompositions/CZFromSinglyControlled.qs:35]: Nit, but I think the Q# language server diagnostics now recommend dropping parens around conditionals in `if` blocks.
```suggestion
            if Length(ctls) == 0 {
```

**Review by cgranade — APPROVED** [review]
LGTM, no substantiative comments.

File: src/Simulation/TargetDefinitions/Decompositions/CX.qs (hunk 1 above)
Comment (swernli) [reply src/Simulation/TargetDefinitions/Decompositions/CX.qs:1]: Good call, I totally spaced on that!

**Pull request merged**